File: 백준/Bronze/2920. 음계/음계.rs
use std::{
    io::{stdin, Read},
    str::FromStr,
};

fn main() {
    let nums = input_vec::<i32>();
    let mut ascending = nums.clone();
    let mut descending = nums.clone();

    ascending.sort();
    descending.sort_by(|a, b| b.cmp(a));

    println!(
        "{}",
        if nums == ascending {
            "ascending"
        } else if nums == descending {
            "descending"
        } else {
            "mixed"
        }
    );
}

fn input_vec<T>() -> Vec<T>
where
    T: FromStr,
    <T as FromStr>::Err: std::fmt::Debug,
{
    let mut buf = String::new();
    stdin().read_to_string(&mut buf).unwrap();

    buf.trim()
        .split_ascii_whitespace()
        .map(|s| s.parse().unwrap())
        .collect::<Vec<T>>()
}

File: 백준/Bronze/2920. 음계/음계.cc
#include <iostream>

using namespace std;

int main() {
  string ans = "mixed";

  int nums[8];

  for (int i = 0; i < 8; ++i) {
    cin >> nums[i];
  }

  for (int i = 2; i < 8; ++i) {
    int curr = nums[i];
    int prev = nums[i - 1];
    int pprev = nums[i - 2];
    if (curr - prev != prev - pprev) {
      break;
    }
    if (i == 7) {
      if (curr - prev == 1) {
        ans = "ascending";
      }
      if (curr - prev == -1) {
        ans = "descending";
      }
    }
  }

  cout << ans;
}